openmls/src/framing/ciphertext.rs

```diff
@@ -445,6 +452,10 @@ impl MlsSenderDataAad {
 /// ```
 #[derive(Debug, Clone)]
 pub(crate) struct MlsCiphertextContent {
+    // The `content` field is serialized and deserialized manually without the
+    // `content_type`, which is not part of the struct as per MLS spec. See the
+    // implementation of `TlsSerialize` for `MlsCiphertextContent`, as well as
+    // `deserialize_ciphertext_content`.
     pub(crate) content: MlsContentBody,
```

Comment:
The other option would be to have a different type here without discriminant and this way avoid manual (de)serialisation. But both ways aren't great. So this is fine.

Reply:
Yes, I think both cases lead to a bit of code duplication. Let's wait and see how often we run into these special cases for serialization/deserialization and then try to come up with a somewhat unified solution.